.github/workflows/reused-commit-msgs-validator.yml

```diff
@@ -8,10 +8,12 @@ on:
 jobs:
   validate-commit-message:
     name: Validate Commit Message
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
+    permissions: 
+      contents: write
     steps:
       - name: Validate Commit Message Content
-        uses: gsactions/commit-message-checker@v1
+        uses: gsactions/commit-message-checker@b88ee88552f16f594ca19cb2c7fcd90df95c9bd4  # v1
```

Comment:
Same question here

Reply:
It has to do with security, given that code attached to a tag can be altered by malicious users whereas commits cannot. [Here is a scientific paper](https://www.usenix.org/conference/usenixsecurity22/presentation/koishybayev) and [a blog post](https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash) about potential security risks related to GitHub Actions

Reply:
This can be mitigated by limiting the set of permissions given to this job, and you have proposed to do the opposite.

Again, the whole purpose of this job is to validate commit messages so they match a particular pattern. Nothing fancy.